core/modules/views/handlers/views_handler_field_numeric.inc

```diff
@@ -124,24 +124,30 @@ class views_handler_field_numeric extends views_handler_field {
       return '';
     }
 
+    // Check to see if hiding should happen before adding prefix and suffix.
+    if ($this->options['hide_empty'] && empty($value) && ($value !== 0 || $this->options['empty_zero'])) {
+      return '';
+    }
+
     if (!empty($this->options['set_precision'])) {
-      $value = number_format($value, $this->options['precision'], $this->options['decimal'], $this->options['separator']);
+      $precision = $this->options['precision'];
+    }
+    elseif ($decimal_position = strpos($value, '.')) {
+      $precision = strlen(rtrim($value, '0')) - $decimal_position - 1;
     }
     else {
-      $remainder = abs($value) - intval(abs($value));
-      $value = $value > 0 ? floor($value) : ceil($value);
-      $value = number_format($value, 0, '', $this->options['separator']);
-      if ($remainder) {
-        // The substr may not be locale safe.
-        $value .= $this->options['decimal'] . substr($remainder, 2);
-      }
+      $precision = 0;
     }
 
-    // Check to see if hiding should happen before adding prefix and suffix.
-    if ($this->options['hide_empty'] && empty($value) && ($value !== 0 || $this->options['empty_zero'])) {
-      return '';
-    }
+    // Use round first to avoid negative zeros.
+    $value = round($value, $precision);
+    // Test against both integer zero and float zero.
+    if ($this->options['empty_zero'] && ($value === 0 || $value === 0.0)) {
+       return '';
+     }
 
+    $value = number_format($value, $precision, $this->options['decimal'], $this->options['separator']);
+    
```

Comment:
Github doesn't show this, but there's trailing white space on this line. :wink:

Reply:
I also can't see it, and no idea how it came in. But somehow I think I removed it.
How did you find it?

Reply:
Confirmed, the extra spaces are gone. :+1: 

You can get a raw diff or commit diff by rewriting the url like:
```
https://patch-diff.githubusercontent.com/raw/backdrop/backdrop/pull/4663.diff
or
https://patch-diff.githubusercontent.com/raw/backdrop/backdrop/pull/4663.patch
```
And then you can apply these patches locally. And that patch caused my local git to nag on the spaces.
Anyway, it's fixed. :smiley: